src/components/atoms/LanguageOption/index.tsx

```diff
@@ -1,16 +1,34 @@
-import * as React from 'react'
-import { StyleLanguageOptionContainer, StyleLanguageOption } from "./styles"
+import React, {useState} from 'react'
+import { StyleLanguageOptionContainer, StyleLanguageOption, StyleLanguageLink } from "./styles"
+import { americanFlagSvg, japanFlagSvg } from "../../../Svgs"
+import { useGetRedirectLanguage } from "../../../../localeRedirect"
 
 interface LanguageOptionInterface {
     language: string
 }
 
+const setLangVer = (props) => {
+    if (props.forcedLanguage === undefined || "ja") {
+        props.forcedLanguage = "en"
```

Comment:
I am almost sure that this is wrong. You're mutating props here.

Reply:
I don't know what else to do though. The automatic redirect immediately returns the user to their browser locale language even if they click the language toggle. 

Even this solution took me days to come up with, so I'm at a complete loss.